docs/ingestion-sink/index.md

```diff
@@ -0,0 +1,224 @@
+# Ingestion Sink
+
+A Java application that runs in Kubernetes, reading input from Google Cloud
+Pub/Sub and emitting records to batch-oriented outputs like GCS or BigQuery.
+Defined in the `ingestion-sink` package
+([source](https://github.com/mozilla/gcp-ingestion/tree/master/ingestion-sink/)).
+
+## Supported Input and Outputs
+
+Supported inputs:
+
+ * Google Cloud PubSub
+
+Supported outputs:
+
+ * Google Cloud PubSub
+ * Google Cloud Storage
+ * Google Cloud BigQuery
+
+### Test Input and Output
+
+Test inputs will stop when an exception is raised or the end of the pipe or
+file is reached.  Supported test inputs:
+
+* `System.in` (stdin), by setting `INPUT_PIPE` to any of `-`, `0`, `in`,
+  `stdin`, `/dev/stdin`
+* A single file, by setting `INPUT_PIPE` to `/path/to/input_file`
+
+Test outputs don't exercise batching and will write messages as newline
+delimited JSON in the order they are received. Supported test outputs:
+
+* `System.out` (stdout), by setting `OUTPUT_PIPE` to any of `-`, `1`, `out`,
+  `stdout`, `/dev/stdout`
+* `System.err` (stderr), by setting `OUTPUT_PIPE` to any of `2`, `err`,
+  `stderr`, `/dev/stderr`
+* A single file, by setting `OUTPUT_PIPE` to `/path/to/output_file`
+
+## Configuration
+
+All configuration is controlled by environment variables.
+
+## Output Specification
+
+Depending on the environment variables provided, the application will
+automatically determine where to deliver messages.
+
+If `OUTPUT_BUCKET` is specified without `BIG_QUERY_OUTPUT_MODE`, then messages
```

Comment:
I know it makes sense to split `BigQuery` into `big_query` using the snake casing rules that we're used to, but for some reason I expect to see this as `BIGQUERY_OUTPUT_MODE`. The python module for BigQuery for example is `google.bigquery`.

Reply:
yeah, that makes sense, and that's why I did it this way. I may work with ops to change it to `BIGQUERY_...` in the future.